Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -2332,14 +2334,15 @@ public void applyReferralCode(final String code, final BranchReferralInitListene
      * user selected clients.</p>
      *
      * @param builder A {@link io.branch.referral.Branch.ShareLinkBuilder} instance to build share link.
+     * @return The instance of the {@link Dialog} holding the share view.
      */
-    private void shareLink(ShareLinkBuilder builder) {
+    private Dialog shareLink(ShareLinkBuilder builder) {
```

Comment:
Ok removed Dialog in return param and added callback for dialog launch and dismiss